### ISSUE TITLE
Updating `as_indexed_json` on after commit update callback

### ISSUE THREAD

**Issue by henrikholm89** [elastic/elasticsearch-rails#721]
I am not sure if it is a bug or intentional. 

However, whenever `__elasticsearch__.update_document` gets triggered, it makes the following call to elasticsearch: `http://localhost:9200/development_users/user/8064/_update`

And this does not seen to trigger the `as_indexed_json` method? 
At least, the following methods on my model/document is not getting updated:

```
  def as_indexed_json(*)
      as_json(
        methods: [:primary_team, :club, :top_player?]
      )
    end
```

However, when I call the `__elasticsearch__.update_document` directly on the model in a rails console, it for some reasons triggers an `http://localhost:9200/development_users/user/8064/`, and this way it gets properly updated. But that looks like an `__elasticsearch__.index_document` http call? 

Is there something I'm missing here?

**Comment by estolfo** [comment]
@henrikholm89 Have you been able to resolve this or is it still an issue?

**Comment by henrikholm89** [comment]
Yes, I have fixed it on my end, but it is still a bug in the framework I guess. 
It is a bit weird that I have to do this

```
  after_commit on: [:update] do
    __elasticsearch__.index_document
  end
```

instead of 

```
  after_commit on: [:update] do
    __elasticsearch__.update_document
  end
```

**Comment by estolfo** [comment]
@henrikholm89 I've tried to reproduce this but am having trouble without a code example. Can you please provide a simple example to demonstrate the issue? Thanks

**Comment by adamcrown** [comment]
I am having the same issue and can confirm that the following code works for me, while `Elasticsearch::Model::Callbacks` does not.
```ruby
  after_commit lambda { __elasticsearch__.index_document  },  on: [:create, :update]
  after_commit lambda { __elasticsearch__.delete_document },  on: :destroy
```

In my case it seems like updates to regular number fields are fine. But updates to `enums` are ignored.

**Comment by estolfo** [comment]
@adamcrown Would you mind providing a code example so I can reproduce this and resolve it? Thanks

**Comment by adamcrown** [comment]
I'd love to, and I probably can at some point. But it won't be for at least a little while. Got a few other high-priority things on my plate right now. But I'll see what I can do. Maybe next week sometime.

**Comment by estolfo** [comment]
@adamcrown Ok, that'd be great. I'll be able to fix this if I can reproduce it and understand exactly what the expected behavior is. Thanks!

**Comment by henrikholm89** [comment]
@estolfo Sorry for the late response. 
However, I just looked through some of the other issues and spotted this one #669 which is the exact same issue as I have. 
I am still not sure if it is a bug or intentional, as the things that are not getting updated in my case, are not properties on the actually model, but an association.

**Comment by abdelbk** [comment]
<code>Elasticsearch::Model::Callbacks</code> still doesn't behave like it should.
